src/wp-cron.php

```diff
@@ -46,6 +38,27 @@
 	require_once __DIR__ . '/wp-load.php';
 }
 
+/** This filter is documented in wp-includes/default-constants.php */
+if ( ! apply_filters( 'wp_cron_endpoint_enabled', true ) ) {
+	if ( ! headers_sent() ) {
+		header( 'X-WP-Cron: Bypass' );
```

Comment:
Instead of a custom header, what about:

```suggestion
		status_header( 403 );
```

Reply:
I'm happy to do this if batcache caches 403 responses (pending adding output to the page, as discussed on the ticket).